src/test/scala/impl/group5/NumberOfLoopsAnalysisTest.scala

```diff
@@ -0,0 +1,80 @@
+package org.tud.sse.metrics
+package impl.group5
+
+import org.scalatest.{FlatSpec, Matchers}
+import testutils.AnalysisTestUtils
+
+import java.io.File
+
+/**
+ * Testet die Klasse NumberOfLooopsAnalysis [[NumberOfLoopsAnalysis]]
+ */
+class NumberOfLoopsAnalysisTest extends FlatSpec with Matchers{
+
+  val fileToTest = new File(getClass.getResource("/demo/pdfbox-2.0.24.jar").getPath)
+  val analysisToTest = new NumberOfLoopsAnalysis()
```

Comment:
Es wird nur mit einer einzigen Testdatei getestet, deren wahre Anzahl an Schleifen unbekannt sein sollte, Es sei denn diese wurden zuvor von Hand gezählt.

Es ist daher ratsam ein paar jar-Dateien zu erstellen über deren Klassenstrukturen man dann genaue Kenntnisse hat.
Hier kann man dann zielgenau verschiedenste Varianten testen

Reply:
Zum testen sollte die anzahl der schleifen in einer Datei abgezählt wurden sein. Was nützt es mir 10 Datein zu prüfen ohne zu wissen ob die Funktion richtig zählt und wie viele Scheifen es sein müssten.

Reply:
Die Frage war genau, ob die Schleifen zum Vergleich von Hand gezählt wurden oder nicht. Das hat sich ja nun geklärt. Sie wurden vonhand gezählt.
Es wäre trotzedem sinvoll weitere Datein zu testen. Am besten sollten die diese Datein selbst erstellt werden und viele verschiedene und verschachteltet Schleifenformen beinhalten.
Ich habe den Quelltext von pdfbox-2.0.24 an den in den Tests genutzten Stellen betrachtet. Er beinhaltet ausschließlich einfache Schleife ohne komplexere Verzweigungen oder Verschachtelungen

Reply:
Neu hinzugefügte Tests: 
- Sehr gute Wahl der Funktionen
- Habe alle Schleifen in diesen Funktionen nachgezählt: alles stimmt
- Sehr gute Arbeit
😃 👍